src/core/message.rs

```diff
@@ -20,4 +21,22 @@ impl message::Message {
             0
         }
     }
+
+    pub fn hex_hash(&self) -> String {
+        hex::encode(&self.hash)
+    }
+}
+
+impl snapchain::ValidatorMessage {
+    pub fn fid(&self) -> u32 {
```

Comment:
Two things:

1) I might prefer something like below for improved readability:

```rust
impl snapchain::ValidatorMessage {
    pub fn fid(&self) -> u32 {
        if let Some(fname) = &self.fname_transfer {
            return fname.fid as u32;
        }
        if let Some(event) = &self.on_chain_event {
            return event.fid as u32;
        }
        0
    }
}
```

I think seeing the unwrap there I really have to scrutinize the code, and it took me a moment to realize that it was indeed safe due to the is_some() above.

I think we can probably remove most of the unwraps() in this file with similar refactors and it will be MUCH faster to reason about.

2) What to do in the fall through case? Should we return an error or even panic? We don't expect this to ever be either? Can we model as actual enum in protobuf?

Reply:
1. Ah, much cleaner. Fixed.
2. I think if we use oneof it will use enums. We should probably switch to that actually. But even if we did, it won't solve the problem because it's acceptable to have 0 as a value in the protobuf, so callers have to handle fid 0 (which is not valid for the application, fids start from 1). So, I think defaulting to 0 is fine here.